api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1301,7 +1304,7 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
 
         if not run_args.field_setup:
             logger.info("Preparing factory profile")
-            for path in ('srv/salt', 'srv/pillar', '.ssh'):
+            for path in ('srv/config', 'srv/salt', 'srv/pillar', '.ssh'):
```

Comment:
I thought by just adding /opt/seagate/cortx/provisioner I can use existing roster and master config as it is , that's why not created dynamically again.

Please suggest me, do I need to create roster and other config again in destroy API with only required pillar and  states path?

Reply:
we copy factory data (setup command input arguments, pillars) just to have the same context for later setups (e.g. when we replace a node): we won't need to ask a user to provide us these values but just reuse the same to get the same logic and result env.

But roster and configs are generated on the fly since they are jump server specific (since paths are there). And it is not a big deal to generate them.

>Please suggest me, do I need to create roster and other config again in destroy API with only required pillar and states path?

if you are asking about initial stage of the destroy logic: possibly. I guess you need that for salt-ssh client (roster and master config). If are sure that all paths inside roster and master config would be ok regtardless of a node where Destory is called then your implementation might work. If you are not sure then you need to create these files on the fly. You may check how ReplaceNode calls SetupProvisioner:
  - Replace [prepares profile](https://github.com/Seagate/cortx-prvsnr/blob/ddaef30c7836f2bbb9fc5c44d812a7a6c1ffd149/api/python/provisioner/commands/replace_node.py#L131-L133) from factory one. 
  - Then it [resolves nodes specs](https://github.com/Seagate/cortx-prvsnr/blob/ddaef30c7836f2bbb9fc5c44d812a7a6c1ffd149/api/python/provisioner/commands/replace_node.py#L192-L203). You may consider to build nodes node in another way actually (you would just need to get IP per miinion id and create a list of `Node` class instances)
  - then roster file can be created : you need to pass only the list of nodes and a path to the key which is `/root/.ssh/id_rsa_prvsnr`

To summarize: to create roster you need to have only a list of nodes. You may actually re-use .ssh/config and build roster from it.

To build master config you should call profile setup API like [here](https://github.com/Seagate/cortx-prvsnr/blob/ddaef30c7836f2bbb9fc5c44d812a7a6c1ffd149/api/python/provisioner/commands/setup_provisioner.py#L1151-L1155).